src/nvim/hardcopy.c

```diff
@@ -3080,16 +3079,18 @@ void mch_print_set_font(int iBold, int iItalic, int iUnderline)
   }
 }
 
-void mch_print_set_bg(long_u bgcol)
+void mch_print_set_bg(uint32_t bgcol)
 {
+  assert(bgcol <= INT_MAX);
```

Comment:
Note that I didn't do all possible type refactoring in these files. There are still a lot of things than can be improved, as I just did the strictly necessary to make -Wconversion pass and remove long_u.
This one is a sensible one, though.